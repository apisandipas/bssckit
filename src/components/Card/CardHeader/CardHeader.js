import styled, { css } from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../../theme';
import { Div } from '../../Utilities/base';

const getBorderRadius = props => {
  if (props.noRadius) {
    return css`
      border-radius: ${theme('cardHeader.borderRadius.noRadius')};
    `;
  }

  return css`
    border-top-left-radius: ${theme('cardHeader.borderRadius.topLeftRight')};
    border-top-right-radius: ${theme('cardHeader.borderRadius.topLeftRight')};
  `;
};

const CardHeader = styled(Div)`
  padding: ${theme('cardHeader.padding.default')};
  background-color: ${theme('cardHeader.colors.default.background')};
  border-bottom: 1px solid ${theme('cardHeader.colors.default.borderBottom')};
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  ${props => getBorderRadius(props)}
`;

CardHeader.defaultProps = {
  theme: defaultTheme,
};

export default CardHeader;
