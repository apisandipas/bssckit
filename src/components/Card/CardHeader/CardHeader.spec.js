import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import CardHeader from './CardHeader';

describe('CardHeader Component', () => {
  it('matches CardHeader', async () => {
    const { container } = render(<CardHeader />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches CardHeader noRadius', () => {
    const { container } = render(<CardHeader noRadius />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });
});
