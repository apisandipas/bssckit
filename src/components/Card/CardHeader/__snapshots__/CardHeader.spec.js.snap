// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader Component matches CardHeader 1`] = `
.c0 {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-top-left-radius: calc(0.25rem - 1px);
  border-top-right-radius: calc(0.25rem - 1px);
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`CardHeader Component matches CardHeader noRadius 1`] = `
.c0 {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;
